host/xtest/asym_cipher_perf.c

```diff
@@ -0,0 +1,854 @@
+	double max;
+	int initialized;
+};
+
+/* Take new sample into account (Knuth/Welford algorithm) */
+static void update_stats(struct statistics *s, uint64_t t)
+{
+	double x = (double)t;
+	double delta = x - s->m;
+
+	s->n++;
+	s->m += delta / s->n;
+	s->m2 += delta * (x - s->m);
+	if (!s->initialized) {
+		s->min = x;
+		s->max = x;
+		s->initialized = 1;
+	} else {
+		if (s->min > x)
+			s->min = x;
+		if (s->max < x)
+			s->max = x;
+	}
+}
+
+static double stddev(struct statistics *s)
+{
+	if (s->n < 2)
+		return NAN;
+
+	return sqrt(s->m2 / s->n);
+}
+
+static void usage(const char *progname, uint32_t width_bits, uint32_t main_algo,
+		  uint32_t mode, uint32_t salt_len, uint32_t size,
+		  uint32_t rsa_algo, int warmup, uint32_t l, uint32_t n)
+{
+	fprintf(stderr, "Usage: %s [-h]\n", progname);
+	fprintf(stderr, "Usage: %s [-t] [-m] [-k SIZE]", progname);
+	fprintf(stderr, " [-t main_algo] [-m mode] [-n LOOP] [-r|--no-inited] [-d WIDTH_BITS]");
+	fprintf(stderr, " [-k SIZE] [-a rsa_algo] [-s salt_len] [-v [-v]] [-w SEC]");
+	fprintf(stderr, "\n");
+	fprintf(stderr, "Asymmetric performance testing tool for OP-TEE\n");
+	fprintf(stderr, "\n");
+	fprintf(stderr, "Options:\n");
+	fprintf(stderr, "  -h|--help        Print this help and exit\n");
+	fprintf(stderr, "  -t TYPE          Test type for asymmetric algo\n");
+	fprintf(stderr, "                   DH, RSA, ECDSA, ECDH, X25519\n");
+	fprintf(stderr, "  -m MODE          Test asymmetric mode\n");
+	fprintf(stderr, "                   ENC, DEC, SIGN, VERIFY, GENKEYPAIR\n");
+	fprintf(stderr, "  -l LOOP          Inner loop iterations [%u]\n", l);
+	fprintf(stderr, "  -n LOOP          Outer test loop iterations [%u]\n", n);
+	fprintf(stderr, "  -r|--random      Get input data from /dev/urandom (default: all zeros)\n");
+	fprintf(stderr, "  -k SIZE          Plaintext Length [%u]\n", size);
+	fprintf(stderr, "  -d WIDTH_BITS    ECC:the width_bits only support 192/224/256/384/521 [%u]\n", width_bits);
+	fprintf(stderr, "                   DH: width_bits <= 2048, RSA: width_bits <= 4096 [%u]\n", width_bits);
+	fprintf(stderr, "  -a ALGO          Crypto algorithm tested when TYPE is RSA\n");
+	fprintf(stderr, "  		    RSA_NOPAD, RSAES_PKCS1_V1_5, RSAES_PKCS1_OAEP_SHA1\n");
+	fprintf(stderr, "         	    RSAES_PKCS1_OAEP_SHA224, RSAES_PKCS1_OAEP_SHA256\n");
+	fprintf(stderr, "         	    RSAES_PKCS1_OAEP_SHA384, RSAES_PKCS1_OAEP_SHA512\n");
+	fprintf(stderr, "  		    RSASSA_PKCS1_V1_5_SHA1, RSASSA_PKCS1_V1_5_SHA224\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_V1_5_SHA256, RSASSA_PKCS1_V1_5_SHA384\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_V1_5_SHA512, RSASSA_PKCS1_PSS_MGF1_SHA1\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_PSS_MGF1_SHA224, RSASSA_PKCS1_PSS_MGF1_SHA256\n");
+	fprintf(stderr, "                   RSASSA_PKCS1_PSS_MGF1_SHA384, RSASSA_PKCS1_PSS_MGF1_SHA512\n");
+	fprintf(stderr, "  -s SALT_LEN      only RSA SSA_PKCS1_PSS support! [%u]\n", salt_len);
```

Comment:
"  -s SALT_LEN      Salt length in bytes (only when ALGO is one of RSASSA_PKCS1_PSS_*) [%u]\n", salt_len